src/server/protocol_client.cc

```diff
@@ -39,6 +50,42 @@ using absl::StrCat;
 
 namespace {
 
+#ifdef DFLY_USE_SSL
+static ProtocolClient::SSL_CTX* CreateSslClientCntx() {
+  const bool is_tls_replication_enabled = absl::GetFlag(FLAGS_tls_replication);
+  if (!is_tls_replication_enabled) {
```

Comment:
I find this flow a bit confusing.
Why not have this check outside of CreateSslClientCntx and call it only if is_tls_replication_enabled